File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> out = new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<>();

        if (root == null) {
            return out; 
        }

        queue.add(root);
        while(!queue.isEmpty()) {
            List<Integer> levelList = new ArrayList<>();   
            Queue<TreeNode> newQueue = new LinkedList<>(); 

            while (!queue.isEmpty()) {
                TreeNode curr = queue.remove();
                levelList.add(curr.val);
                if (curr.left != null) {
                    newQueue.add(curr.left);
                }
                if (curr.right != null) {
                    newQueue.add(curr.right);
                }
            }
            out.add(levelList);
            queue = newQueue;
        }

        return out;
    }
}

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:

        def cmp(p, q):

            if p is None:
                return p == q
            if q is None:
                return q == p
            if p.val != q.val:
                return False

            return cmp(p.left, q.left) and cmp(p.right, q.right)
        
        def sub(p, q):

            if not q:
                return True
            if not p:
                return False
            if cmp(p, q):
                return True

            return sub(p.left, q) or sub(p.right, q)
        
        return sub(root, subRoot)